server/middleware/registration/index.js

```diff
@@ -0,0 +1,104 @@
+const bcrypt = require('bcrypt');
+const logger = require('simple-node-logger').createSimpleLogger();
+const {
+    HttpStatusCodes: { StatusCodes },
+    setLogDetails,
+    DataException
+} = require('../../utils/constants');
+const { REQUEST_TIMEOUT, OK, BAD_REQUEST, GATEWAY_TIMEOUT } = StatusCodes
+const mongoUtil = require('../../utils/mongoUtil');
+const { HttpStatusCodes } = require('../../utils/constants');
+const JWT = require('jsonwebtoken');
+const { uuid } = require('uuidv4');
+
+const registrationService = async (req, res) => {
+    const saltRounds = 10;
+    let data, statusCode, userId, collectionObj;
+
+    const {
+        firstName,
+        lastName,
+        prefLoc,
+        title,
+        email,
+        userName,
+        password,
+    } = req.body;
+    userId = email;
```

Comment:
refactor to:

`const {
        firstName,
        lastName,
        prefLoc,
        title,
        email: userId,
        userName,
        password,
    } = req.body;`

Reply:
actually, let's do this another way...

`const { email: userId, password, ...rest } = req.body`